neo4j/src/main/java/org/hibernate/ogm/datastore/neo4j/dialect/impl/Neo4jAssociationSnapshot.java

```diff
@@ -63,6 +68,28 @@ public int size() {
 	}
 
 	private static Iterable<Relationship> relationships(Node ownerNode, AssociationKey associationKey, String relationshipType) {
+		if ( relationshipType.contains( "." ) ) {
+			String[] pathToAssociation = EMBEDDED_FIELDNAME_SEPARATOR.split( relationshipType );
+			Node nextNode = ownerNode;
+			for ( int i = 0; i < pathToAssociation.length; i++ ) {
+				String splitType = pathToAssociation[i];
```

Comment:
Make sense, althought I suspect this logic should not be here in the first place. I see what I can do